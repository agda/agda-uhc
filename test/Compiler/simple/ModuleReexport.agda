module ModuleReexport where

open import Common.Unit
open import Common.Nat
open import Common.IO

module A (B : Set) (b : B) where
  data X : Set where
    Con1 : B -> X
    Con2 : X
  f : X -> B
  f (Con1 x) = x
  f Con2 = b


module X = A Nat 10


main = printNat (A.f Nat 10 (X.Con1 20)) ,,
  putStrLn "" ,,
  printNat (A.f Nat 10 X.Con2) ,,
  putStrLn "" ,,
  return unit
