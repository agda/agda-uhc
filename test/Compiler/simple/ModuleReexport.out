EXECUTED_PROGRAM

ret > ExitSuccess
out > 20
out > 10
out >
